### PR TITLE
Separate out StablePoolAmplification tests from StablePhantomPool tests

### PR DESCRIPTION
**Pull request by TomAFrench**
Making a draft PR to help avoid duplication of work. Currently it uses the full pool contract but I'll make a mock before we merge.

This PR moves tests for StablePoolAmplification functionality into a separate file so that we can ensure full coverage of this in isolation from the rest of the pool

### CODE REVIEW

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+
+describe('StablePoolAmplification', () => {
+  let owner: SignerWithAddress, admin: SignerWithAddress, other: SignerWithAddress;
+  let pool: Contract;
+
+  const AMP_PRECISION = 1e3;
+  const AMPLIFICATION_PARAMETER = bn(200);
```
Comment (nventuro) [inline pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts:24]:
```suggestion
  const INITIAL_AMPLIFICATION_PARAMETER = bn(200);
```

Comment (nventuro) [reply pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts:24]: We should also have the tests for initial values (checked in the constructor) here

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+        context('when requesting a valid amp', () => {
+          const itUpdatesAmpCorrectly = (newAmp: BigNumber) => {
+            const increasing = AMPLIFICATION_PARAMETER.lt(newAmp);
+
+            context('when there is no ongoing update', () => {
+              it('starts changing the amp', async () => {
+                await pool.connect(owner).startAmplificationParameterUpdate(newAmp, endTime);
+
+                await advanceTime(duration / 2);
```
Comment (nventuro) [inline pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts:57]: Can we advance to any point other than the midpoint? :upside_down_face: e.g. one third

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        });
+      });
+    });
+
+    context('when the sender is not allowed', () => {
+      it('reverts', async () => {
+        await expect(pool.connect(other).stopAmplificationParameterUpdate()).to.be.revertedWith('SENDER_NOT_ALLOWED');
+        await expect(
+          pool.connect(other).startAmplificationParameterUpdate(AMPLIFICATION_PARAMETER, DAY)
+        ).to.be.revertedWith('SENDER_NOT_ALLOWED');
```
Comment (nventuro) [inline pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts:243]: Do we not have any tests for setting this via the Authorizer for delegated owners?

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+      });
+    });
+  });
+
+  describe('stopAmplificationParameterUpdate', () => {
```
Comment (nventuro) [inline pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts:248]: We're lacking permission tests for this

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts (hunk 3 above)
Comment (TomAFrench) [reply pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts]: [8ef8496](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1559/commits/8ef84963b0e759e2f9dddde50b3d8ebbeb926f24)

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts (hunk 4 above)
Comment (TomAFrench) [reply pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts:286]: [d9a297a](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1559/commits/d9a297ac6257f8a7fc34a019b8e9e6c602b5f6b1)

File: pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts (hunk 1 above)
Comment (TomAFrench) [reply pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts]: [a67ebca](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1559/commits/a67ebcae8e1f824a824edf3642c879f24bb9bed8)

**Review by nventuro — APPROVED** [review]
Looks great! There's also amp constructor tests in the original file that we could remove.

**Pull request merged**